src/Compiler/lex.fsl

```diff
@@ -845,41 +846,43 @@ rule token args skip = parse
 
  | ">]" { GREATER_RBRACK }
 
- | "{"
+ | ("{{" | "{")
      {
+       let isDouble = lexbuf.LexemeLength > 1
        match args.stringNest with
-       | [] -> ()
-       | (counter, style, m) :: rest ->
+       | (counter, style, isDoubleDollar, m) :: rest when isDoubleDollar = isDouble ->
          // Note, we do not update the 'm', any incomplete-interpolation error
          // will be reported w.r.t. the first '{'
-         args.stringNest <- (counter + 1, style, m) :: rest
+         args.stringNest <- (counter + 1, style, isDoubleDollar, m) :: rest
+       | _ -> ()
        // To continue token-by-token lexing may involve picking up the new args.stringNes
        let cont = LexCont.Token(args.ifdefStack, args.stringNest)
        LBRACE cont
      }
 
  | "|" { BAR }
 
- | "}"
+ | ("}}" | "}")
```

Comment:
This is parsing "}}" as a single "}" in regular F# code, hence the tests failing to compile

Reply:
You can probably add a new RBRACE_RBRACE token and smash it apart in the lexfilter. Though be careful of |}} too and check if anything else can be combined with }. 

Likewise LBRACE of course

Reply:
Sample smash is here:

https://github.com/dotnet/fsharp/blob/main/src/Compiler/SyntaxTree/LexFilter.fs#L2442

Not sure there's any better way to do it

Reply:
@dsyme As I don't know well how the lexfilter works and as you say, we can have problems with things like `{{|`, I'm trying to [move back the lexeme end position](https://github.com/dotnet/fsharp/pull/13259/commits/57e81fb713c88a44fd029adbcabda12037c312c7) if we shouldn't consume the two braces. Is this a bad idea?

Also, for some reason CI is not running tests. I cannot get repo to build locally either. When I run `Build.cmd` at one point I get these error (same in `main` branch):

```
C:\Users\alfon\repos\fsharp\tests\FSharp.Core.UnitTests\FSharp.Core\RefStructs.fs(20,18): error FS0412: A type instanti
ation involves a byref type. This is not permitted by the rules of Common IL. [C:\Users\alfon\repos\fsharp\tests\FSharp
.Core.UnitTests\FSharp.Core.UnitTests.fsproj]
C:\Users\alfon\repos\fsharp\tests\FSharp.Core.UnitTests\FSharp.Core\RefStructs.fs(20,6): error FS0437: A type would sto
re a byref typed value. This is not permitted by Common IL. [C:\Users\alfon\repos\fsharp\tests\FSharp.Core.UnitTests\FS
harp.Core.UnitTests.fsproj]
```

Reply:
CI is not running tests if there's a merge conflict, IIRC.
The local issue is weird, does this happen if you build with `-noVisualStudio` switch too?

Reply:
Thanks a lot @vzarytovskii! I will fix the merge conflicts 👍 And the `-noVisualStudio` switch does indeed work! I do have Visual Studio installed but I don't use it a lot and maybe the installation is faulty (I had problems to open FSharp.sln with VS).

Reply:
> Thanks a lot @vzarytovskii! I will fix the merge conflicts 👍 And the `-noVisualStudio` switch does indeed work! I do have Visual Studio installed but I don't use it a lot and maybe the installation is faulty (I had problems to open FSharp.sln with VS).

Huh, interesting, thanks. I will add some notes to the DEVGUIDE on how to deal with such errors (incl. VS repair and building with the flag).